persistence/remote/src/main/java/org/infinispan/persistence/remote/upgrade/HotRodTargetMigrator.java

```diff
@@ -126,7 +128,11 @@ public long synchronizeData(Cache<Object, Object> cache, int readBatch, int thre
                for (List<Integer> partition : partitions) {
                   Set<Integer> segmentSet = new HashSet<>();
                   segmentSet.addAll(partition);
-                  futures.add(executor.submit(iterator.next(), new MigrationTask(segmentSet, readBatch, threads)));
+                  DistributedTask<Integer> task = executor
+                     .createDistributedTaskBuilder(new MigrationTask(segmentSet, readBatch, threads))
+                     .timeout(0, TimeUnit.NANOSECONDS)
```

Comment:
AFAIK the fact that 0 means infinite timeout is not documented and might be removed in the future, I'd suggest using `Long.MAX_VALUE` instead.